pom.xml

```diff
@@ -47,29 +48,40 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <functional.java>4.5</functional.java>
+        <java.level>6</java.level>
```

Comment:
According to my yesterdays chat with @mamalisk, core should be Java6 compatible. Hence this change and not using diamond syntax, replacing it with explicit types declaration. 
Having things like ArrayList<Any> doesn't make code java 8 incompatible.